File: Bosetterne/Bosetterne-service/src/main/resources/WebContent/ressurser/dart/login/login_token.dart

import 'dart:html';
import 'dart:convert';
import 'package:html5lib/parser.dart' show parse;
import 'package:html5lib/dom.dart';


class Login_token {
  List<AnchorElement> _menyLinker;
  String _access_token;
  Oauth2Cred _autorisering;

  Login_token() {
    if(window.localStorage["bosetterne_token"] != null) {
      Map token = JSON.decode(window.localStorage["bosetterne_token"]);
      _access_token = token['access_token'];
      _autorisering = new Oauth2Cred(_access_token);
    }
  }

  String get access_token => _access_token == null ? "" :_access_token;

  /** Plukker opp alle menylenkene og Legger ved access token hvis den eksisterer.
   * Hvis access_ token ikke eksisterer, legges den ikke ved.
   *
   */
  void menyLenker() {
    if(_access_token != null) {
      _menyLinker = querySelectorAll('a');
      _menyLinker.forEach((a) => _setToken(a));
    }
  }

  void _setToken(AnchorElement anchorElement) {

    print(anchorElement.getAttribute('href'));
    StringBuffer tokenQuery = new StringBuffer();
    tokenQuery.write(anchorElement.getAttribute('href'));
    tokenQuery.write("?token=");
    tokenQuery.write(_access_token);

    anchorElement.setAttribute('href',tokenQuery.toString());
    print(anchorElement.getAttribute('href'));
  }
}

class Oauth2Cred {
  String token;

  Oauth2Cred(this.token);

  String toString() {
    return 'Bearer ' + window.btoa(token);
  }
}

File: Brettspill/Brettspill-service/src/main/resources/WebContent/app/js/modules/LoginToken.js
/**
 * Created by Espen on 27.04.2015.
 */
/*global define*/

var $ = require("jquery");

export default class LoginToken {
    constructor() {

        var a = $("#loggut");
        if(a.length){
            console.log(a);

            var token = window.sessionStorage.getItem('bosetterne_token');
            if(token) {
                token = JSON.parse(token);
                this._access_token = token.access_token;
                this._autorisering = new Oauth2Cred(this._access_token);
            } else {
                this._access_token = '';
            }
            
            
            this.menyLenker();
            a.click(function(){
                window.sessionStorage.removeItem("bosetterne_token");
            });
        } else {
            window.sessionStorage.removeItem("bosetterne_token");
            console.log("token fjernet");
        }
    }

    /** Plukker opp alle menylenkene og Legger ved access token hvis den eksisterer.
     * Hvis access_ token ikke eksisterer, legges den ikke ved.
     */
    menyLenker() {
        if(this._access_token) {
            this._menyLinker = $('a.meny');
            this._menyLinker.each(function (index, value){
                this._setToken($(this));
            });
        } else {
            console.log("Ikke innlogget");
        }
    };

    _setToken(anchorElement) {
        var link = anchorElement.getAttribute('href');
        var queryPos = link.indexOf('?');
        if (queryPos > -1) {
            link = link.substring(0, queryPos);
        }
        var tokenQuery = link+"?innlogget_token="+this._access_token;

        anchorElement.attr('href',tokenQuery);
    };

    get access_token() {
        return this._access_token;
    };
}

class Oauth2Cred {

    constructor(token) {
        this.token = token;
    }

    toString() {
        return 'Bearer ' + window.btoa(this.token);
    }
}